pkg/rmb/rmb.go

```diff
@@ -0,0 +1,251 @@
+	m.handlers[topic] = handler
+}
+
+// Run runs listeners to the configured handlers
+// and will trigger the handlers in the case an event comes in
+func (m *MessageBus) Run() error {
+	con := m.pool.Get()
+	defer con.Close()
+
+	topics := make([]string, len(m.handlers))
+	for topic := range m.handlers {
+		topics = append(topics, topic)
+	}
+
+	jobs := make(chan Message, numWorkers)
+	for i := 1; i <= numWorkers; i++ {
+		go m.worker(context.Background(), jobs)
+	}
+
+	for {
+		if m.Context.Err() != nil {
+			return nil
+		}
+
+		data, err := redis.ByteSlices(con.Do("BLPOP", redis.Args{}.AddFlat(topics).Add(0)...))
+		if err != nil {
+			log.Err(err).Msg("failed to read from system local messagebus")
+			return err
+		}
+
+		var message Message
+		err = json.Unmarshal(data[1], &message)
+		if err != nil {
+			log.Err(err).Msg("failed to unmarshal message")
+			continue
+		}
+
+		_, ok := m.handlers[string(data[0])]
+		if !ok {
+			log.Debug().Msg("handler not found")
+			continue
+		}
+
+		jobs <- message
+	}
+}
+
+func (m *MessageBus) worker(ctx context.Context, jobs chan Message) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case message := <-jobs:
+			bytes, err := message.GetPayload()
+			if err != nil {
+				log.Err(err).Msg("err while parsing payload reply")
+			}
+
+			handler, ok := m.handlers[message.Command]
+			if !ok {
+				log.Warn().Msg("handler not found")
+			}
+
+			var tKey twinKeyID
+			ctx = context.WithValue(ctx, tKey, message.TwinSrc)
+
+			var mKey messageKey
+			ctx = context.WithValue(ctx, mKey, message)
+
+			data, err := handler(ctx, bytes)
+			if err != nil {
+				log.Err(err).Msg("err while handling job")
+				// TODO: create an error object
+				message.Err = err.Error()
+			}
+
+			err = m.SendReply(message, data)
+			if err != nil {
+				log.Err(err).Msg("err while sending reply")
+			}
+		}
+	}
+}
+
+func GetMessage(ctx context.Context) (*Message, error) {
+	message, ok := ctx.Value(messageKey{}).(Message)
+	if !ok {
+		return nil, errors.New("failed to load message from context")
+	}
+
+	return &message, nil
+}
+
+// SendReply send a reply to the message bus with some data
+func (m *MessageBus) SendReply(message Message, data interface{}) error {
+	con := m.pool.Get()
+	defer con.Close()
+
+	// reply to source
+	message.TwinDest = []uint32{message.TwinSrc}
+
+	// base 64 encode the response data
+	// message.Data = base64.StdEncoding.EncodeToString(data)
+
+	// set the time to now
+	message.Epoch = time.Now().Unix()
+
+	bytes, err := json.Marshal(message)
+	if err != nil {
+		return err
+	}
+
+	_, err = con.Do("RPUSH", replyBus, bytes)
+	if err != nil {
+		log.Err(err).Msg("failed to push to reply messagebus")
+		return err
+	}
+
+	return nil
+}
+
+// PushMessage pushes a message to a topic
+// for testing purposes
+func (m *MessageBus) PushMessage(topic string, message Message) error {
```

Comment:
Same here, this doesn't have to be public

Reply:
for testing purposes right now